src/metax_api/services/catalog_record_service.py

```diff
@@ -526,27 +528,115 @@ def validate_reference_data(cls, research_dataset, cache):
             raise ValidationError(errors)
 
     @classmethod
-    def strip_catalog_record(cls, catalog_record):
+    def remove_contact_info_metadata(cls, rd):
         """
-        Strip the catalog record of any confidential/private information not supposed to be
-        available for the general public.
+        Strip research dataset of any confidential/private contact information not supposed to be available for the
+        general public
 
-        Accepts as paramater either a single cr object (dict), or a list of objects.
+        :param rd:
+        :return: research dataset with removed contact information
+        """
+        return remove_keys_recursively(rd, ['email', 'telephone', 'phone'])
+
+    @classmethod
+    def check_and_remove_metadata_based_on_access_type(cls, rd):
+        """
+        If necessary, strip research dataset of any confidential/private information not supposed to be
+        available for the general public based on the research dataset's access type.
+
+        :param rd:
+        :return: research dataset with removed metadata based on access type
+        """
 
-        :param catalog_record:
+        access_type_id = cls.get_research_dataset_access_type(rd)
+        if access_type_id == ACCESS_TYPES['open']:
+            pass
+        elif access_type_id == ACCESS_TYPES['embargoed']:
+            try:
+                access_rights_available = cls.get_research_dataset_embargo_available(rd)
+                embargo_time_passed = now_is_later_than_datetime_str(access_rights_available)
```

Comment:
I think the new comparison function is unnecessary. There are already utils ``parse_timestamp_string_to_tz_aware_datetime()`` and ``get_tz_aware_now_without_micros()``. The new function ``now_is_later_than_datetime_str()`` seems to do some stuff with timezones when getting current time, but probably the existing function should then be altered (or parameterized), if the timezone-trickery is really necessary.